### PR TITLE
Abstract cuDNN from Pooling implementation

### PR DESCRIPTION
**Pull request by mrwyattii**
Builds off work in #1638 and #1643 to abstract the cuDNN API calls out of the Pooling layer. Depends on #1642, needs to be rebased after #1642 is merged.

### CODE REVIEW

File: include/lbann/utils/dnn_lib/cudnn/pooling.hpp
Comment (timmoon10) [inline include/lbann/utils/dnn_lib/cudnn/pooling.hpp:41]:
```suggestion
void pooling_forward(PoolingDescriptor const& poolingDesc,
```

File: include/lbann/utils/dnn_lib/cudnn/pooling.hpp
Comment (timmoon10) [inline include/lbann/utils/dnn_lib/cudnn/pooling.hpp:65]:
```suggestion
void pooling_forward(PoolingDescriptor const& poolingDesc,
```

File: include/lbann/utils/dnn_lib/cudnn/pooling.hpp
Comment (timmoon10) [inline include/lbann/utils/dnn_lib/cudnn/pooling.hpp:82]:
```suggestion
void pooling_backward(PoolingDescriptor const& poolingDesc,
```

File: include/lbann/utils/dnn_lib/cudnn/pooling.hpp
Comment (timmoon10) [inline include/lbann/utils/dnn_lib/cudnn/pooling.hpp:114]:
```suggestion
void pooling_backward(PoolingDescriptor const& poolingDesc,
```

**Review by timmoon10 — CHANGES_REQUESTED** [review]
LGTM after merging the suggestions.

**Review by timmoon10 — APPROVED** [review]
:shipit:

**Review by benson31 — APPROVED** [review]
LGTM

**Pull request merged**
